tests/api/test_order.py

```diff
@@ -2997,22 +3034,79 @@ def test_update_invoice_missing_number(
     }
-    response = user_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
+    response = staff_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
     assert_no_permission(response)
 
 
-def test_update_invoice_invalid_id(user_api_client, permission_manage_orders):
+def test_update_invoice_invalid_id(staff_api_client, permission_manage_orders):
     variables = {"id": "SW52b2ljZToxMzM3", "number": "01/12/2020/TEST"}
-    user_api_client.user.user_permissions.add(permission_manage_orders)
-    response = user_api_client.post_graphql(UPDATE_INVOICE_MUTATION, variables)
+    response = staff_api_client.post_graphql(
+        UPDATE_INVOICE_MUTATION, variables, permissions=[permission_manage_orders]
+    )
     content = get_graphql_content(response)
     errors = content["data"]["updateInvoice"]["invoiceErrors"][0]
     assert errors["code"] == InvoiceErrorCode.NOT_FOUND.name
     assert errors["field"] == "id"
+
+
+def test_create_invoice(staff_api_client, permission_manage_orders, orders):
+    number = "01/12/2020/TEST"
+    url = "http://www.example.com"
+    order = orders[0]
+    variables = {
+        "orderId": graphene.Node.to_global_id("Order", order.pk),
+        "number": number,
+        "url": url,
+    }
+    response = staff_api_client.post_graphql(
+        CREATE_INVOICE_MUTATION, variables, permissions=[permission_manage_orders]
+    )
+    content = get_graphql_content(response)
+    invoice = Invoice.objects.get(order_id=order.pk, url=url, number=number)
+    assert invoice.url == content["data"]["createInvoice"]["invoice"]["url"]
+    assert invoice.number == content["data"]["createInvoice"]["invoice"]["number"]
+    assert invoice.status == InvoiceStatus.READY
+
+
+def test_create_invoice_invalid_id(staff_api_client, permission_manage_orders):
+    variables = {
+        "orderId": graphene.Node.to_global_id("Order", 1337),
+        "number": "01/12/2020/TEST",
+        "url": "http://www.example.com",
+    }
+    response = staff_api_client.post_graphql(
+        CREATE_INVOICE_MUTATION, variables, permissions=[permission_manage_orders]
+    )
+    content = get_graphql_content(response)
+    errors = content["data"]["createInvoice"]["invoiceErrors"][0]
+    assert errors["code"] == InvoiceErrorCode.NOT_FOUND.name
+    assert errors["field"] == "orderId"
+
+
+def test_create_invoice_empty_params(
+    staff_api_client, permission_manage_orders, orders
+):
+    order = orders[0]
+    variables = {
+        "orderId": graphene.Node.to_global_id("Order", order.pk),
+        "number": "",
+        "url": "",
+    }
+    response = staff_api_client.post_graphql(
+        CREATE_INVOICE_MUTATION, variables, permissions=[permission_manage_orders]
+    )
+    content = get_graphql_content(response)
+    errors = content["data"]["createInvoice"]["invoiceErrors"]
+    for field in ("number", "url"):
+        assert_field_has_error(field, InvoiceErrorCode.REQUIRED.name, errors)
```

Comment:
A better option is to use assets here instead of putting them into function. We can do this way.
```
    assert errors[0] == {
        "field": "number",
        "code": InvoiceErrorCode.REQUIRED.name,
    }
    assert errors[1] == {
        "field": "url",
        "code": InvoiceErrorCode.REQUIRED.name,
    }
```